packages/teleterm/src/ui/AppInitializer.tsx

```diff
@@ -1,29 +1,32 @@
-import React, { FC, useEffect } from 'react';
-
-import { useAsync } from 'shared/hooks/useAsync';
+import React, { FC, useCallback, useEffect, useState } from 'react';
 import styled from 'styled-components';
 
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 
 export const AppInitializer: FC = props => {
   const ctx = useAppContext();
-  const [state, init] = useAsync(() => ctx.init());
-
-  useEffect(() => {
-    init();
-  }, []);
+  const [isUiVisible, setIsUiVisible] = useState(false);
+
+  const initializeApp = useCallback(async () => {
+    try {
+      await ctx.runBeforeUiIsVisible();
+      setIsUiVisible(true);
+      await ctx.runAfterUiIsVisible();
```

Comment:
Previously, in `init()` method we had some code that blocked it execution (`await this.clustersService.syncRootClusters()`) and also rendering the app.

For asking about collecting usage metrics I wanted to do something similar. But, then I realized that it can't show a dialog while there is no UI 😏. I decided to split init function into two: before and after the UI is visible. It allows me to show the dialogs in a correct order: first collecting data, then restoring workspace, etc.

Reply:
I decided to revert the change to `AppContext`, the code that is ran after showing UI is now placed in `initUi.ts`, so we have less code in the context.